File: courses/lesson5_cars/src/main/java/by/academy/lesson5/cars/AddCarActivity.java
package by.academy.lesson5.cars;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import static android.view.View.INVISIBLE;
import static by.academy.lesson5.cars.MainActivity.ADD;
import static by.academy.lesson5.cars.MainActivity.COMMAND;
import static by.academy.lesson5.cars.MainActivity.ITEM;

public class AddCarActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.car_add_edit);
        findViewById(R.id.removeBUtton).setVisibility(INVISIBLE);


        TextView ownerView = findViewById(R.id.viewTextOwnerName);
        TextView producerView = findViewById(R.id.viewTextProducer);
        TextView modelView = findViewById(R.id.viewTextModel);
        TextView plateNumberView = findViewById(R.id.viewTextPlateNumber);



        findViewById(R.id.addBUtton).setOnClickListener(view -> {
            DataItem dataItem = new DataItem(
                    String.valueOf(ownerView.getText()),
                    String.valueOf(producerView.getText()),
                    String.valueOf(modelView.getText()),
                    String.valueOf(plateNumberView.getText())
            );

            Intent data = new Intent();
            data.putExtra(COMMAND, ADD);
            data.putExtra(ITEM, dataItem);
            setResult(RESULT_OK, data);
            finish();
        });


        findViewById(R.id.backButton).setOnClickListener(view -> {
            setResult(RESULT_CANCELED);
            finish();

        });
    }
}

File: courses/lesson5_cars/src/main/java/by/academy/lesson5/cars/AddCarActivity.kt
package by.academy.lesson5.cars

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import by.academy.lesson5.cars.data.CarInfoEntity

class AddCarActivity : AppCompatActivity() {

    private var mCurrentPhotoPath: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.car_add_edit)

        findViewById<View>(R.id.removeBUtton).visibility = View.INVISIBLE

        val ownerView = findViewById<TextView>(R.id.viewTextOwnerName)
        val producerView = findViewById<TextView>(R.id.viewTextProducer)
        val modelView = findViewById<TextView>(R.id.viewTextModel)
        val plateNumberView = findViewById<TextView>(R.id.viewTextPlateNumber)


        findViewById<View>(R.id.addBUtton).setOnClickListener {
            val dataItem = CarInfoEntity(0L,
                    ownerView.text.toString(),
                    producerView.text.toString(),
                    modelView.text.toString(),
                    plateNumberView.text.toString(),
                    mCurrentPhotoPath
            )

            val data = Intent().apply {
                action = MainActivity.ADD
                putExtra(MainActivity.ITEM, dataItem)
            }
            setResult(RESULT_OK, data)
            finish()
        }
        findViewById<View>(R.id.backButton).setOnClickListener {
            setResult(RESULT_CANCELED)
            finish()
        }
    }
}